src/GafferImage/Merge.cpp

```diff
@@ -278,7 +278,7 @@ IECore::ConstFloatVectorDataPtr Merge::merge( F f, const std::string &channelNam
 			continue;
 		}
 
-		IECore::ConstStringVectorDataPtr channelNamesData = (*it)->channelNamesPlug()->getValue();
+		IECore::ConstStringVectorDataPtr channelNamesData = (*it)->channelNames();
```

Comment:
Same here - can we share a GlobalContext with the dataWindow access below? Just to be clear, not by adding a getChannelNamesAndDataWindow accessor on ImagePlug, just by manually making a GlobalContext here.